src/Brick/Main.hs

```diff
@@ -242,19 +247,19 @@ customMainWithVty :: (Ord n)
                   -- loop (you write to this channel, the event loop reads from
                   -- it). Provide 'Nothing' if you don't plan on sending custom
                   -- events.
-                  -> App s e n
+                  -> App s e n m
                   -- ^ The application.
                   -> s
                   -- ^ The initial application state.
-                  -> IO (s, Vty)
+                  -> m (s, Vty)
 customMainWithVty initialVty buildVty mUserChan app initialAppState = do
     let run vty rs st brickChan = do
             result <- runWithVty vty brickChan mUserChan app rs st
-                `E.catch` (\(e::E.SomeException) -> shutdown vty >> E.throw e)
+                -- `E.catch` (\(e::E.SomeException) -> shutdown vty >> E.throw e)
```

Comment:
This helps restore the normal vty mode on exception.   Did you have something elsewhere that was handling this instead?

Reply:
I didn't, but it's trivial to switch it into a `MonadCatch`. 

I needed to generalize the underlying monad for a side project, and thought I should upstream the results. I'd be happy to make this PR less shit if the maintainers are interested --- but if it's a definite "no merge" I didn't want to waste my time.